codegen/type_converter.go

```diff
@@ -69,24 +69,101 @@ func (c *TypeConverter) getGoTypeName(valueType compile.TypeSpec) (string, error
 	return GoType(c.Helper, valueType)
 }
 
+//   input of "A.B.C.D" returns ["A","A.B", "A.B.C", "A.B.C.D"]
+func getMiddleIdentifiers(identifier string) []string {
+	subIds := strings.Split(identifier, ".")
+
+	middleIds := make([]string, 0, len(subIds))
+
+	middleIds = append(middleIds, subIds[0])
+	for i := 1; i < len(subIds); i++ {
+		middleIds = append(middleIds, fmt.Sprintf("%s.%s", middleIds[i-1], subIds[i]))
+	}
+
+	return middleIds
+}
+
+//  converts a list of identifier paths into boolean nil check expressions on those paths
+func convertIdentifiersToNilChecks(identifiers []string) []string {
+	checks := make([]string, 0, len(identifiers)-1)
+
+	for i, ident := range identifiers {
```

Comment:
why not `identifiers[1:]`

Reply:
it can potentially be out of range

Reply:
I had a size check on identifiers here but test/coverage indicated that it was unreachable.

cli's suggestion is kind of a style vs performance nit . depending on how smart the compiler is, identifies[1:] might allocate a new slice wrapper unnecessarily.

Reply:
i had a size check on identifiers but testing/coverage indicated it was unreachable

i think cli's suggestion is sort of a style vs performance nit.  unless the compiler is super smart (?),  `identifiers[1:]`  will allocate a new slice wrapper unnecessarily